faunadb-common/src/main/java/com/faunadb/common/Connection.java

```diff
@@ -356,32 +360,84 @@ public void syncLastTxnTime(long newTxnTime) {
     return performRequest("PATCH", path, Optional.of(body), Map.of(), queryTimeout);
   }
 
-  public static URI appendUri(URI oldUri, String queryKey, String queryValue) throws URISyntaxException {
+  private static URI appendUri(URI oldUri, String queryKey, List<String> queryValues) throws URISyntaxException {
     String urlEncodedKey = URLEncoder.encode(queryKey, StandardCharsets.UTF_8);
-    String urlEncodedValue = URLEncoder.encode(queryValue, StandardCharsets.UTF_8);
+    String urlEncodedValue = queryValues.stream()
+            .map(u -> URLEncoder.encode(u, StandardCharsets.UTF_8))
+            .collect(Collectors.joining(","));
     String query = urlEncodedKey + "=" + urlEncodedValue;
     return new URI(oldUri.getScheme(), oldUri.getAuthority(), oldUri.getPath(),
             oldUri.getQuery() == null ? query : oldUri.getQuery() + "&" + query, oldUri.getFragment());
   }
 
   private CompletableFuture<HttpResponse<String>> performRequest(String httpMethod, String path, Optional<JsonNode> body,
-                                                                 Map<String, String> params, final Optional<Duration> requestQueryTimeout) throws Exception {
+                                                                 Map<String, List<String>> params, final Optional<Duration> requestQueryTimeout) throws Exception {
```

Comment:
Is it at all possible to narrow the `throws` declaration? It's overly broad as-is. Or even better, since this method returns a completable future, not throw an exception at all.

Reply:
I am proposing the following fix for this https://github.com/fauna/faunadb-jvm/pull/272/commits/93326b2be47d45f52ad05040f1c581164fe96066

@lregnier some of it could have landed directly in https://github.com/fauna/faunadb-jvm/pull/270 I think

Reply:
Yeah, perfect! I was thinking of doing something similar as well (catch and create failed `CompletableFutures`). We should move that to the other PR.

Reply:
I think the blocks could probably be piped together nicer and in a more functional way with `CompletableFuture.thenCompose`, but migrating `faunadb-common` to Scala would probably be time better spent as well.

Reply:
@lregnier I am not really familiar with the Java FP flavor sorry :smile: 

@freels WDYT about this change, is this enough to unblock the PR?